install/helm/agones/templates/crds/fleet.yaml

```diff
@@ -110,6 +110,22 @@ spec:
                           anyOf:
                             - type: integer
                             - type: string
+                priorities: # which gameservers in the Fleet are most important to keep around - impacts scale down logic. Must be identical to priorities in gameserverset.yaml.
+                  type: array
+                  title: Configuration of Counters and Lists scale down logic
+                  nullable: true
+                  items:
+                    type: object
+                    properties:
+                      priorityType:
```

Comment:
We decided to name it priorityType because `type` is a key word as in `type: object`.

I can't remember why we didn't do an enum, maybe for verbose failure? If it's better to do an enum I can switch those over. Would we still use string for `order` so that we can do defaulting? Or would that be better as an enum as well and require the user to specify the order?

Reply:
So on `type` - I'm looking at this and it looks totally inconsistent to me, so I think we should change it. Sorry about the backtrack.

Example:
https://github.com/googleforgames/agones/blob/cef40425076e5e9a766952eb2f499d9b447b703a/install/helm/agones/templates/crds/fleetautoscaler.yaml#L59

I think it's better to move validation into the CRD spec as much as possible, as long term we'll want to not have webhooks as much as we can (move to CEL etc), as it's easier to scale and should be more performant as well.

Also we use an enum and default it within a CRD if it's blank in the Fleet autoscaler as well, so it should be fine (we can also make it `nullable` if it becomes tricky I think).

For example:

https://github.com/googleforgames/agones/blob/cef40425076e5e9a766952eb2f499d9b447b703a/install/helm/agones/templates/crds/fleetautoscaler.yaml#L62-L69

Reply:
Does `priorityType` really need to be named `type`? I would argue that line 63 should have been named `policyType` rather than `type`. Yes, it's technically possible, but it makes the CRD confusing to read since we're overloading a field name with specific meaning in the OpenAPISchema rather than being precise (or less confusingly generic). Same thing with Go, which has `type` as a reserved word. We can (and did) use `Type` as a field name, but that feels about as off as using `Func` as a function name.

Reply:
So I'm not fussed about the Go struct side (in fact k8s also keeps a Go struct name that isn't `type`.

But `type` in the CRD is what K8s use, and what we use in a variety of places, so this is inconsistent with the rest of the API surface.

Here's the K8s example in code:

https://github.com/googleforgames/agones/blob/cef40425076e5e9a766952eb2f499d9b447b703a/vendor/k8s.io/api/apps/v1/types.go#L396-L408

https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy

So yeah, I think it does to be consistent with both K8s and the rest of our CRD api surface.

Reply:
Ah naming, I'll change it over to Type.

Reply:
😆 I know, right?